controllers/utils.js

```diff
@@ -33,71 +33,161 @@ function hasUserSignedUpForRelay(user) {
   return false;
 }
 
-function getExperimentBranch(req, sorterNum = false, language = false) {
+function chooseVariation(variations, sorterNum){
 
-  if (sorterNum === false) {
-    sorterNum = Math.floor(Math.random() * 100);
-    log.debug("No coinflip number provided. Coinflip number is ", sorterNum);
+  let totalPercentage = 0;
+
+  // calculate and store total percentage of variations
+  for (const v in variations) {
+    if (variations.hasOwnProperty(v) && typeof variations[v] === "number") {
+      // multiply by 100 to allow for percentages to the hundredth
+      // (and avoid floating point math errors)
+      totalPercentage += (variations[v] * 100);
+    }
+  }
+
+  totalPercentage = totalPercentage / 100;
```

Comment:
Looks like you could refactor this logic into a `const totalPercentage = getTotalPercentage(variations)` helper function?

Reply:
Filed! https://github.com/mozilla/blurts-server/issues/1739